src/inheritAttributes.js

```diff
@@ -132,16 +132,20 @@ export const getSegmentInformation = (adaptationSet) => {
  *         Callback map function
  */
 export const inheritBaseUrls =
-(adaptationSetAttributes, adaptationSetBaseUrls, periodAdaptationSetInfo) => (representation) => {
+(adaptationSetAttributes, adaptationSetBaseUrls, adaptationSetSegmentInfo) => (representation) => {
   const repBaseUrlElements = findChildren(representation, 'BaseURL');
   const repBaseUrls = buildBaseUrls(adaptationSetBaseUrls, repBaseUrlElements);
   const attributes = shallowMerge(adaptationSetAttributes, getAttributes(representation));
-  const segmentInfoFromRepresenation = getSegmentInformation(representation);
-  const segmentRepresentationInfo = merge(periodAdaptationSetInfo, segmentInfoFromRepresenation);
+  const representationSegmentInfo = getSegmentInformation(representation);
 
   return repBaseUrls.map(baseUrl => {
     return {
-      segmentInfo: segmentRepresentationInfo,
+      segmentInfo: {
+        list: adaptationSetSegmentInfo.list && representationSegmentInfo.list ? merge(adaptationSetSegmentInfo.list, representationSegmentInfo.list) : adaptationSetSegmentInfo.list && !representationSegmentInfo.list ? adaptationSetSegmentInfo.list : !adaptationSetSegmentInfo.list && representationSegmentInfo.list ? representationSegmentInfo.list : undefined,
```

Comment:
This is a nightmare to read. I think its best we try to come at this with a different approach.

The main issue with the merge is that if the second object in the merge has a property that is `undefined`, it overwrites that property with `undefined`, so removing those properties from the object before we even get to merging should solve the issue without needing all this complex logic.

I think that we should return to what you had before this commit of just merging the full segment info objects instead of merging these properties individually.

Then, to fix the issue of a property being `undefined`, in `getSegmentInformation`, before returning the segment info object, save it to a variable and then delete properties that are undefined
```
const segmentInfo = {
  template,
  timeline: ...
  ....
};

Object.keys(segmentInfo).forEach(key => {
  if (!segmentInfo[key]) {
    delete segmentInfo[key];
  }
});

return segmentInfo;
```